File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/controller/PurchaseController.java
package com.rad.fairyteller.controller;

import com.rad.fairyteller.domain.Purchase;
import com.rad.fairyteller.service.PurchaseService;
import com.rad.fairyteller.service.dto.PurchaseDto;
import com.rad.fairyteller.service.mapper.PurchaseMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@CrossOrigin
@RequiredArgsConstructor
public class PurchaseController {

    private final PurchaseService purchaseService;
    private final PurchaseMapper purchaseMapper;

    @RequestMapping(path = "/purchaseListByUserId/{id}")
    public List<PurchaseDto> getPurchaseListByUserId(@PathVariable final Long id) {
        return purchaseMapper.toDto(purchaseService.getPurchaseListByUserId(id));
    }

    @PostMapping(path = "/purchase/save")
    public void saveOrUpdatePurchase(@RequestBody final Purchase purchase) {
        purchaseService.saveOrUpdatePurchase(purchase);
    }
}

File: Sources/Server/fairy-teller/src/main/java/com/rad/fairyteller/controller/PurchaseController.kt
package com.rad.fairyteller.controller

import com.rad.fairyteller.domain.purchase.Purchase
import com.rad.fairyteller.service.PurchaseService
import com.rad.fairyteller.service.dto.PurchaseDto
import com.rad.fairyteller.service.mapper.PurchaseMapper
import org.springframework.web.bind.annotation.*

@RestController
class PurchaseController(private val purchaseService: PurchaseService) {

    @RequestMapping(path = arrayOf("/purchaseListByUserId/{id}"))
    fun getPurchaseListByUserId(@PathVariable id: Long): List<PurchaseDto> {
        return PurchaseMapper.INSTANCE.toDto(purchaseService.getPurchaseListByUserId(id))
    }

    @RequestMapping(path = arrayOf("/purchase/save"), method = arrayOf(RequestMethod.POST))
    fun saveOrUpdatePurchase(@RequestParam purchase: Purchase) =
            purchaseService.saveOrUpdatePurchase(purchase)

    @RequestMapping(path = arrayOf("/purchase/save"))
    fun savePurchase() {
        val purchase = PurchaseMapper.INSTANCE.toEntity(PurchaseDto(102, 2L, 1L))
        purchaseService.saveOrUpdatePurchase(purchase)
    }
}
